tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -123,29 +125,31 @@ def test_it_doesnt_add_report_submission_config_if_lms_not_canvas(
         assert "submissionParams" not in context.js_config.config
 
     def test_it_configures_client_to_focus_on_user_if_in_canvas_and_param_set(
-        self, context, canvas_request, h_api
+        self, context, pyramid_request, h_api
     ):
+        context.is_canvas = True
```

Comment:
This kind of feels like a step backwards because of the duplication, but on the other hand it's very clear about what makes this tick. So on balance I think I like the explicit nature of this in the class where some of these are canvas calls and some aren't.

Reply:
I've replaced this with a fixture, so it how works like this:

```python
    @pytest.mark.usefixtures("is_canvas")
    def test_it_configures_client_to_focus_on_user_if_in_canvas_and_param_set(
        self, context, pyramid_request, h_api
    ):
        ...

...

@pytest.fixture
def is_canvas(context):
    """Set the LMS that launched us to Canvas."""
    context.is_canvas = True
```

Tests still have to explicitly declare that they want Canvas with `@pytest.mark.usefixtures("is_canvas")` but the definition of what constitutes "is canvas" is no longer duplicated, and more test set up is moved out of tests and into fixtures